### PR TITLE
instrument storage query usage

### PR DESCRIPTION
**Pull request by tanabarr**
Build-only: true

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

### CODE REVIEW

File: src/control/server/ctl_ranks_rpc.go [hunk 1]
```diff
@@ -295,30 +296,47 @@ func (svc *ControlService) ResetFormatRanks(ctx context.Context, req *ctlpb.Rank
...
 	if err != nil {
 		return nil, err
 	}
 
+	nvmeScanResp, err := svc.NvmeScan(storage.BdevScanRequest{})
+	if err != nil {
+		svc.log.Errorf("nvme scan failed: %s", err)
+		nvmeScanResp = &storage.BdevScanResponse{}
+	}
+	if nvmeScanResp == nil {
```
Comment (knard-intel) [inline src/control/server/ctl_ranks_rpc.go:309]: Do not see, how nil could be returned without an error.
I am probably missing something?
Maybe to support future backend provider?

File: src/control/server/instance_drpc.go [hunk 2]
```diff
@@ -215,86 +216,83 @@ func (ei *EngineInstance) getSmdDetails(smd *ctlpb.SmdDevResp_Device) (*storage.
...
 }
 
 // updateInUseBdevs updates-in-place the input list of controllers with new NVMe health stats and
 // SMD metadata info.
 //
 // Query each SmdDevice on each I/O Engine instance for health stats and update existing controller
 // data in ctrlrMap using PCI address key.
-func (ei *EngineInstance) updateInUseBdevs(ctx context.Context, ctrlrMap map[string]*storage.NvmeController) error {
+func (ei *EngineInstance) updateInUseBdevs(ctx context.Context, ctrlrMap map[string]*storage.NvmeController) (err error) {
+	defer func() {
```
Comment (knard-intel) [inline src/control/server/instance_drpc.go:234]: nice :)

**Review by knard-intel — APPROVED** [review]
Seems OK for me.
But not sure, of all the side effects which could happen with the refactoring of the updateInUseBdevs function.

**Comment by tanabarr** [comment]
replaced by https://github.com/daos-stack/daos/pull/8062 which should be ported to release/2.0 branch after lambing to master

**Pull request closed without merging**